iac-extensions/arm/src/test/java/org/sonar/iac/arm/parser/ArmJsonBaseConverterTest.java

```diff
@@ -165,4 +170,51 @@ void shouldShiftTextRangesOfExpressionNodes() {
     ArmTreeAssert.assertThat(stringLiteral)
       .hasRange(1, 8, 3, 15);
   }
+
+  @Test
+  void shouldParseMultilineString() {
+    var code = """
+      {
+        "function": "foo
+          bar"
+      }
+      """;
+    var parser = new ArmParser();
+    var file = (FileImpl) parser.parse(code, null);
+    var multilineString = (ScalarTree) file.document().elements().get(0).value();
+    assertThat(multilineString.value()).isEqualTo("foo\n    bar");
+  }
+
+  @Test
+  void shouldConvertMultilineFunctionCall() {
+    var code = """
+      {
+        "function": "[function_call(
+          'foobar'
+        )]"
+      }
+      """;
+    var parser = new ArmParser();
+    var file = (FileImpl) parser.parse(code, null);
+    var functionScalar = (ScalarTree) file.document().elements().get(0).value();
+    var converter = new ArmJsonBaseConverter(null);
+    var expression = converter.toExpression(functionScalar);
+
+    assertThat(expression.getKind()).isEqualTo(ArmTree.Kind.FUNCTION_CALL);
+
+    var functionCall = ((FunctionCall) expression);
+    assertThat(functionCall.textRange()).hasRange(2, 14, 4, 5);
+
+    var identifier = functionCall.name();
+    assertThat(identifier.value()).isEqualTo("function_call");
+    assertThat(identifier.textRange()).hasRange(2, 15, 2, 28);
+
+    var arguments = functionCall.argumentList().elements();
+    assertThat(arguments).hasSize(1);
+
+    var argument = arguments.get(0);
+    assertThat(argument.getKind()).isEqualTo(ArmTree.Kind.STRING_LITERAL);
+    assertThat(argument).hasRange(3, 19, 3, 27);
```

Comment:
And here for sure the values are wrong as line 3  has only 12 characters. IMO the shifting without converting first to Location (or pointer offset from the beginning of the file) doesn't work. I had the same problem in my PR and it is a why I did this conversion to Location.